### PR TITLE
(Leave for after Launch) Server file upload - size validation

### LINKED ISSUE #1256: Restrict file upload by file extension and size.
## Summary

We are using FRS as a baseline for file upload.
We should have the same upload restrictions: jpg, jpeg, doc, docx, xls xlsx, pdf, txt, rtf. File size restriction: 4MB

## Motivation

Limiting the amount of stuff we can receive at the start. We can iterate on this and adjust as needed.

## Unresolved questions
FRS also limits a user to 3 uploads per complaint, but is this necessary? 
How many 4MB files can we fit in an e-mail through RCMP mail gateway?
OCRE allows pngs, I don't think it would hurt to add pngs to start?

### PR DESCRIPTION
**Pull request by naomiseminega**
Fixes #1256 

# Description
-Added file size validation on Server side, to catch any file bigger than 4MB
- Server won't allow saving the file if it's bigger than 4MB limit
- It will log warning

# Any new packages installed?
No
> Give details

# Checklist:
- [x] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [x] I have added and needed tests for my changes (in particular for new screens)
- [x] I have added a comment to any confusing code

# Required followup work
No

### CODE REVIEW

File: f2/server.js [hunk 1]
```diff
@@ -54,10 +58,13 @@ const uploadData = async (req, res, fields, files) => {
...
-  // Await here because we also need these results before saving
-  await scanFiles(data)
+  // Check file size
+  if (checkFileSize(files)) {
+    // Await here because we also need these results before saving
+    await scanFiles(data)
 
-  contentModeratorFiles(data, () => save(data, res))
+    contentModeratorFiles(data, () => save(data, res))
+  }
```
Comment (sastels) [inline f2/server.js]: Running locally It hangs if I add a file - I think we need some sort of 
```
else {
  res.statusCode = 502
   res.statusMessage = 'Error: files too big'
   res.send(res.statusMessage)
}
```
here

Comment (naomiseminega) [reply f2/server.js]: Fixed!

**Comment by sastels** [comment]
Running locally, if I do basically a blank report without files I get 
```
Your reference number:
Error: files too big
```

**Comment by naomiseminega** [comment]
> Running locally, if I do basically a blank report without files I get
> 
> ```
> Your reference number:
> Error: files too big
> ```
> 
> oh! I think it's because `CheckFileSize()` returns true if and only if the files are too big, and in `server.js` we scan and save if `CheckFileSize()` is true.
> 
> Probably best to rename the function something like `FilesAreTooBig()` or something to make it clear what `true` and `false` mean 😄

Oh :p ! I remember I got the same bug last time (remember issue  #1421 (?), server error was being printed on the thank you page. I'll try your suggestion.

**Comment by sastels** [comment]
Will resolve in #1648 instead (using a different approach since we now are dealing with files slightly differently on the server)

**Pull request closed without merging**